pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go

```diff
@@ -43,7 +49,9 @@ var _ = Describe("VMI Stats Collector", func() {
 	clusterConfig, _, _ = testutils.NewFakeClusterConfigUsingKV(&k6tv1.KubeVirt{})
 
 	Context("VMI info", func() {
-		setupTestCollector()
+		BeforeEach(func() {
+			setupTestCollector()
+		})
```

Comment:
Does this fix a bug?

Reply:
Does it?

Reply:
No, the `ctrl := gomock.NewController(GinkgoT())`  requires it.

```
DeferCleanup must be called inside a setup or subject node
vendor/github.com/golang/mock/gomock/controller.go:137
  You must call DeferCleanup inside a setup node (e.g. BeforeEach, BeforeSuite,
  AfterAll...) or a subject node (i.e. It).  You can't call DeferCleanup at the
  top-level or in a container node - use the After* family of setup nodes
  instead.
 ```

Reply:
This smells, when we need to change unit tests because of refactoring.